cli/cli.go

```diff
@@ -439,123 +447,135 @@ func getMigrateSubcommand(subcmd, usage string) cli.Command {
 			cli.StringFlag{Name: "config-file", Value: defaultConfigFile, Usage: "Server config File"},
 			cli.BoolFlag{Name: lockWithEtcd, Usage: "Enable if ETCD should be used to synchronize migrations"},
 		},
-		Action: migrationSubcommandWithLock(func(context *cli.Context) {
+		Action: getLockedMigrationAction(func(context *cli.Context) {
 			if migration.Run(subcmd, context.Args()) != nil {
 				os.Exit(1)
 			}
 		}),
 	}
 }
 
-func getMigrateSubcommandWithPostMigrateEvent(subcmd, usage string) cli.Command {
-	const (
-		PostMigrationEventTimeoutFlag = "post-migration-event-timeout"
-		ConfigFileFlag                = "config-file"
-		EmitPostMigrationEventFlag    = "emit-post-migration-event"
-		PostMigrationEvent            = "post-migration"
-		SyncEtcdEventFlag             = "sync-etcd-event"
-	)
-	return cli.Command{
-		Name:  subcmd,
-		Usage: usage,
-		Flags: []cli.Flag{
-			cli.StringFlag{Name: ConfigFileFlag, Value: defaultConfigFile, Usage: "Server config File"},
-			cli.BoolFlag{Name: EmitPostMigrationEventFlag, Usage: "Enable if post-migration event should be emitted to modified schema extensions"},
-			cli.DurationFlag{Name: PostMigrationEventTimeoutFlag, Value: time.Second * 30, Usage: "Maximum duration of post-migration event"},
-			cli.BoolFlag{Name: SyncEtcdEventFlag, Usage: "Enable if ETCD events should be synchronized after migration"},
-			cli.BoolFlag{Name: lockWithEtcd, Usage: "Enable if ETCD should be used to synchronize migrations"},
-		},
-		Action: migrationSubcommandWithLock(func(context *cli.Context) {
-			config := util.GetConfig()
+func actionMigrate(subcmd string, context *cli.Context) {
+	if migration.Run(subcmd, context.Args()) != nil {
+		os.Exit(1)
+	}
+}
 
-			if migration.Run(subcmd, context.Args()) != nil {
-				os.Exit(1)
-			}
+func actionPostMigrateEvent(context *cli.Context) {
```

Comment:
I know that this code was already here, but I think it does not belong here: should be moved from "cli" package to a separate "migration" package

Reply:
ok, fixed

Reply:
Moved to a separate file. Unfortunately, moving it out of cli package is difficult because of the circular dependencies. It could be fixed by adding some interfaces but this is not worth the effort.